assets/src/edit-story/masks/display.js

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+
+import StoryPropTypes from '../types';
+import { useTransformHandler } from '../components/transform';
+import { getDefinitionForType } from '../elements';
+import { getElementMask } from './';
+
+const FILL_STYLE = {
+  position: 'absolute',
+  top: 0,
+  left: 0,
+  right: 0,
+  bottom: 0,
+};
+
+export default function WithMask({
+  element,
+  fill,
+  style,
+  children,
+  box,
+  ...rest
+}) {
+  const [replacement, setReplacement] = useState(null);
+  const mask = getElementMask(element);
+  useTransformHandler(element.id, (transform) => {
+    const replacementElement = transform?.updates?.replaceElement;
+    if (replacementElement) {
+      const { type } = replacementElement;
+      const { Display } = getDefinitionForType(type);
+      setReplacement(<Display element={replacementElement} box={box} />);
```

Comment:
IMHO, we should use `setReplacement` to store the actual `replacementElement` and do the switch in the rendering below (or `displayElement.js` as mentioned above).